src/main/java/com/google/sps/data/FirestoreDB.java

```diff
@@ -85,86 +130,161 @@ public long voteRecipe(String Id, int voteDiff) {
         break;
     }
     return DBUtils.blockOnFuture(commentsQuery.get()).toObjects(Comment.class);
-  }
+  }  
 
+  /**
+   * Returns all the tags in the DB.
+   * @param getHidden if true, get all the tags in the DB. If false, only get non-hidden tags.
+   * @return list of tags
+   */
   public List<Tag> getAllTags(boolean getHidden) {
     Query tagsQuery =
         getHidden ? DBUtils.tags() : DBUtils.tags().whereEqualTo(Tag.HIDDEN_KEY, false);
     return DBUtils.blockOnFuture(tagsQuery.get()).toObjects(Tag.class);
   }
   
-  /** Returns a User object from userId. */
-    public User getUser(String userId) {
-      DocumentReference userRef = DBUtils.user(userId);
-      DocumentSnapshot user = DBUtils.blockOnFuture(userRef.get());
-      return user.toObject(User.class);
-    }
-   
-    /** Adds a User to the db based on an input userId. */
-    public String addUser(String userId) {
-      DocumentReference newUserRef = DBUtils.users().document(userId);
-      User newUser = new User(userId);
-      DBUtils.blockOnFuture(newUserRef.set(newUser));  // .set() returns an ApiFuture
-      return userId;
-    }
-    
-    /** Delete a User from the db based on an input userId. */
-    public void deleteUser(String userId) {
-      DBUtils.blockOnFuture(DBUtils.user(userId).delete());
-    }
+  /** 
+   * Determines if a document exists in a Firestore collection.
+   * @param docId the document's Firebase ID.
+   * @param collection the collection the document should be in; MUST BE A COLLECTION CONSTANT from DBUtils.java
+   * @return false if the ID is not associated with any doc in the specified collection, 
+   * true if it is.
+   */
+  public boolean isDocument(String docId, String collection) {
+    DocumentReference docRef;
     
-    /**
-     * Sets a property's value to "true" for a certain user document.
-     * Can be used to let a user add a recipe to saved or created, or to let user follow a tag.
-     * @param userId the user's Firebase ID
-     * @param objectId the ID of either a recipe if the intent is to save/create, or of a tag for tag following.
-     * @param collection a KEY constant from User class indicating which mode -- save, create, or follow tag.
-     */
-    public void makeUserPropertyTrue(String userId, String objectId, String collection) {
-      DocumentReference userRef = DBUtils.user(userId);
-      String nestedPropertyName = DBUtils.getNestedPropertyName(collection, objectId);
-      ApiFuture addUserPropertyFuture = 
-          userRef.update(Collections.singletonMap(nestedPropertyName, true));
-      DBUtils.blockOnFuture(addUserPropertyFuture);
+    // Look in the appropriate bin.
+    switch (collection) {
+      case DBUtils.DB_RECIPES_COLLECTION:
```

Comment:
Just make the Firestore public and access this using `DBUtils.database.collection(collection)`
Honestly DBUtils is becoming more and more of a wrapper class around the Firestore and at this point bypassing it for readability is worth it I think.